### ISSUE TITLE
Can someone please compile a stand-alone RUST version for macOS of the latest gnirehtet?

### ISSUE THREAD

**Issue by LeeBinder** [Genymobile/gnirehtet#479]
The last pre-compiled version for macOS was[gnirehtet-rust-macos64-v2.2.1.zip](https://github.com/Genymobile/gnirehtet/releases/download/v2.2.1/gnirehtet-rust-macos64-v2.2.1.zip) (old release). The brew install of v.2.5 requires the entire rust package be installed, which "weighs" 912 MB!

The stand-alone gnirehtet-rust-macos64-v2.2.1 gnirehtet binary got the few rust libs/ deps which gnirehtet relies upon integrated during compile, which has it "weigh" only 1.1 MB. In other words, brew installs more than 900 MB of rusty junk, at least in regards to gnirehtet..

If someone knows how to build a stand-alone RUST version for macOS, please share. @rom1v I'm aware you set this project to inactive recently, but if you were so kind and share the steps how to build a Mac version which does not rely on tons of unused deps, it would be really appreciated :) 👍 

Thank you.

**Comment by VaradGupta23** [comment]
Steps to Compile a Standalone gnirehtet Rust Binary for macOS
 1. Install Required Tools
Make sure you have:

macOS (x64 or ARM depending on your chip)

Rust toolchain via [rustup](https://rustup.rs/)


curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh
Install cargo dependencies:


rustup target add x86_64-apple-darwin
 2. Clone the gnirehtet repo

git clone https://github.com/Genymobile/gnirehtet.git
cd gnirehtet
3. Modify Cargo.toml (optional for size reduction)
You can enable LTO and strip symbols to minimize the binary size:


[profile.release]
opt-level = "z"
lto = true
codegen-units = 1
panic = "abort"

 4. Build a standalone static binarybash

cargo build --release
This will output a binary in:

arduino

target/release/gnirehtet
 5. Strip symbols (optional but reduces size)

strip target/release/gnirehtet
Expected final size: ~1.1–1.8 MB depending on Rust version and optimizations.

 6. Package (Optional)
Zip it:


cd target/release
zip gnirehtet-rust-macos64.zip gnirehtet
 Notes:
This method does not require the 900MB+ Rust environment at runtime.

The Rust toolchain is only required once during build.

The resulting binary can be used standalone on any other macOS system.

TL;DR One-liner (after setting Rust)

git clone https://github.com/Genymobile/gnirehtet.git && cd gnirehtet && cargo build --release && strip targ

**Comment by LeeBinder** [comment]
Neat - thank you, @VaradGupta23. Is the resulting binary (much) smaller than the build compiled via GitHub CI this repos's author provides in the releases section?
